hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -379,6 +401,17 @@ void pgraph_write(void *opaque, hwaddr addr, uint64_t val, unsigned int size)
             qemu_cond_broadcast(&pg->flip_3d);
         }
         break;
+    case NV_PGRAPH_RDI_DATA: {
+        unsigned int select = GET_MASK(pg->regs[NV_PGRAPH_RDI_INDEX],
+                                       NV_PGRAPH_RDI_INDEX_SELECT);
+        unsigned int address = GET_MASK(pg->regs[NV_PGRAPH_RDI_INDEX],
+                                        NV_PGRAPH_RDI_INDEX_ADDRESS);
+        pgraph_rdi_write(pg, select, address, val);
+        assert(address < NV_PGRAPH_RDI_INDEX_ADDRESS); /* FIXME: Overflow into select? */
```

Comment:
This probably doesn't work due to the shift, it should probably be something like `< GET_MASK(NV_PGRAPH_RDI_INDEX_ADDRESS, NV_PGRAPH_RDI_INDEX_ADDRESS)` to extract the number of bits